### ISSUE TITLE
deal.II 8.4.0 release

### ISSUE THREAD

**Issue by tjhei** [dealii/dealii#1938]
I would like to start the discussion about the next release. I am not sure we can manage another holiday release, but we can at least aim for it. :-)

Thinks we need to think about:
- Windows support? Ready in time?
- Should we remove the experimental new linear algebra vectors?
- Fix currently broken unit tests
- Some extensive testing with intel compilers and on Mac OS
- make linux installer based on candi official?

Other open things I am missing?

**Comment by Rombur** [comment]
I think we also need to find a way to advertise the code gallery and make sure that everything works with the new release.

**Comment by drwells** [comment]
There are a few things on my todo list I should be able to handle by the end of the year:

- [ ] Finish the 'regularization' of type names
- [ ] fix #1691 (I have a patch, I just need to write tests). @tjhei is this necessary for Windows support?

**Comment by tjhei** [comment]
> @tjhei is this necessary for Windows support?

No. If you clone the git repository on windows it will convert the line endings correctly and parse them. It doesn't hurt though if people mix things when transferring parameter files manually.

**Comment by luca-heltai** [comment]
Other points to take care of:

- [ ] Remove deprecated functions in Triangulation::set/get boundary related to Manifolds
- [ ] Break backward compatibility with boundary ids

**Comment by bangerth** [comment]
May I advocate that we postpone the work on boundaries and manifolds until just after the release? I'm not quite convinced that we can get this right within a couple of weeks before a release. We also have a few other incompatible things we could all roll into a major 9.0 release after the upcoming one.

**Comment by luca-heltai** [comment]
Sure!

**Comment by tjhei** [comment]
I was thinking about going through the first steps of the release process and branch off of master. Anything big still coming that I am missing?

**Comment by davydden** [comment]
@tjhei it would be good to finish with https://github.com/dealii/dealii/pull/1804

**Comment by davydden** [comment]
wrt intel compilers, i have a number of tests timed out and a couple failed on `module load gcc/4.9.2 intel64/15.0up02` :
```
TIMEOUT:
	3137 - codim_one/grid_refinement.release (Timeout)
	3791 - fe/fe_values_view_02.release (Timeout)
	3883 - fe/injection_dgq.release (Timeout)
	4209 - grid/grid_generator_01a.debug (Timeout)
	4272 - grid/grid_in_msh_version_2.release (Timeout)
	4445 - grid/mesh_3d_6.debug (Timeout)
	4603 - hp/create_laplace_matrix_02b.release (Timeout)
	4605 - hp/create_laplace_matrix_03.release (Timeout)
	4622 - hp/create_mass_matrix_02b.debug (Timeout)
	4632 - hp/create_mass_matrix_05.debug (Timeout)
	4648 - hp/dof_renumbering_04.debug (Timeout)
	4735 - hp/hp_constraints_q_system_x.release (Timeout)
	6017 - mpi/distort_random_01.mpirun=8.debug (Timeout)
	7006 - numerics/no_flux_hp_04.release (Timeout)

FAILED	
	848 - base/anisotropic_1.release (Failed) -- indeed fails
	4355 - grid/grid_tools_local_to_global_vertex_id.mpirun=2.debug (Failed)
	4356 - grid/grid_tools_local_to_global_vertex_id.mpirun=2.release (Failed)
	5331 - lac/linear_operator_08.debug (Failed)
	5332 - lac/linear_operator_08.release (Failed)
```
any1 tried running the test-suite with Intel?

I have a feeling that some of those time-outs are related to extensive output. Interestingly enough, on the same claster GCC+OpenMPI is good.

**Comment by tjhei** [comment]
@davydden I am running a nightly tester using intel 15: http://cdash.kyomu.43-1.org/viewTest.php?onlyfailed&buildid=278

**Comment by davydden** [comment]
@tjhei in an attempt to have less things screwed up i just re-installed deal.ii with candi and `gcc/4.9.2 intel64/15.0up05` and have insane amount of tests fail... For the results you showed, what's the gcc version you used? Perhaps some combinations do not play together well...

**Comment by tjhei** [comment]
gcc 4.6.3. can you submit your test results to the cdash?

**Comment by davydden** [comment]
I actually just wiped the build it as it was really a lot, like half of the tests... I will re-install with `gcc/4.8.1` and same intel versions to see if that changes anything... In any case, could you please remind me how does one submit results to cdash?

**Comment by tjhei** [comment]
https://www.dealii.org/developer/developers/testsuite.html#submit

**Comment by tjhei** [comment]
What do we do about the new Vector classes? I think our options are:
a) leave them in as is
b) move them into an ``experimental`` namespace
c) remove them completely

thoughts?

**Comment by bangerth** [comment]
And/or merge #1901 !

**Comment by Rombur** [comment]
Let's not merge it. I am still working on it. In December, we realized that we forgot about some functionality in the new interface. I have been working on this but I have no test and I used C++11 (I plan to change that when I am sure the code works).

**Comment by tjhei** [comment]
@Rombur what do you propose we do with the current files then? Can we just remove them for the release?

**Comment by Rombur** [comment]
@tjhei Yeah that's fine.

**Comment by tjhei** [comment]
I am through with the release steps and I just pushed a pre-release at https://github.com/dealii/dealii/releases/tag/v8.4.0-rc2

The only thing left is the announcement and release paper, I think.

**Comment by tjhei** [comment]
With all the changes we just made, I think we need another RC. I will do that later.

**Comment by tamiko** [comment]
> With all the changes we just made, I think we need another RC.

Would be nice to have :-)

**Comment by tjhei** [comment]
done, working on offline-doc now.

**Comment by tjhei** [comment]
@luca-heltai can you make another mac bundle please?

**Comment by luca-heltai** [comment]
Sure. I'm on it.

**Comment by tjhei** [comment]
@luca-heltai The latest .dmg on a macbook with el capitan has the problem that "make run" doesn't work (can't find the .lib) while running the program directly works. Should we set the DY_LIB_PATH or has it something to do with the SIP setting?

**Comment by tjhei** [comment]
Note that dealii-8.3.0-el-capitan.dmg is setting a DYLD_LIB_PATH and there "make run" is working.

**Comment by tjhei** [comment]
@luca-heltai  ?

Can anybody else please test the OSX builds? I would prefer to bring out the release in the next couple of days. 

@bangerth Should we just go ahead and ignore OSX for now?

**Comment by davydden** [comment]
@tjhei i will run unit tests on os-x...

**Comment by davydden** [comment]
@tjhei or did you mean the .dmg builds?

**Comment by tjhei** [comment]
yes, I am talking about the .dmg package, see above.

**Comment by luca-heltai** [comment]
I'm rebuilding everything. On my system, everything seems to run fine, so I don't know where the problem may be... and I haven't changed SIP.

**Comment by davydden** [comment]
i tested the .dmg build and also have a problem with running:
```
dyld: Library not loaded: /lib/libdeal_II.g.8.4.0-rc3.dylib
  Referenced from: /Users/davydden/Desktop/work/C++/dealii-tmp/2015-09-25-step-18/./step-18
  Reason: image not found
```

**Comment by luca-heltai** [comment]
I've just prepared a new package. Can anyone with a mac test this one? @davydden ?

**Comment by luca-heltai** [comment]
If this one works, I'll get back the DYLD_LIBRARY_PATH in the configuration file. I guess having it won't hurt, anyway...

**Comment by luca-heltai** [comment]
I remember adding to the `setup_cached_variables.cmake` a variation of the following, which forces global paths upon install, but I no longer find this in history. @tamiko , do you think this would be possible?

~~~
  SET_TARGET_PROPERTIES(${_lib} PROPERTIES MACOSX_RPATH OFF
    BUILD_WITH_INSTALL_RPATH OFF
    INSTALL_NAME_DIR "${CMAKE_INSTALL_PREFIX}/${D2K_LIBRARY_RELDIR}"
    ARCHIVE_OUTPUT_DIRECTORY "${CMAKE_BINARY_DIR}/${D2K_LIBRARY_RELDIR}"
    LIBRARY_OUTPUT_DIRECTORY "${CMAKE_BINARY_DIR}/${D2K_LIBRARY_RELDIR}"
    RUNTIME_OUTPUT_DIRECTORY "${CMAKE_BINARY_DIR}/${D2K_EXECUTABLE_RELDIR}"
   )
~~~

**Comment by davydden** [comment]
@luca-heltai that one worked!

**Comment by luca-heltai** [comment]
:+1:

I'm not happy about that one, because it uses DYLD_LIBRARY_PATH. This will give problems, for example for user codes compiled in XCode, which does not respect DYLD_LIBRARY_PATHS set in the terminal. 

The final solution should be the one above, but I'm not confident that this won't break anything for other people without a mac. @tamiko , opinions?

**Comment by tjhei** [comment]
> but I'm not confident that this won't break anything for other people without a mac

Wrap it with ``IF(CMAKE_SYSTEM_NAME MATCHES "Darwin")``?

**Comment by bangerth** [comment]
On a different note, I've read through the release paper again and made sure it's in good shape. It is now here, and also linked to from the release announcement, so at least this part of the release is good to go:
  https://www.dealii.org/deal84-preprint.pdf

**Comment by tjhei** [comment]
okay, are we ready to do tag the final release? @tamiko @bangerth @luca-heltai ?

**Comment by tamiko** [comment]
@tjhei I don't see any obstacles for tagging a release. Looks fine from my side.

**Comment by tamiko** [comment]
@bangerth We could also mention the Tensor interface/implementation cleanup in section 2.2 :-)

**Comment by bangerth** [comment]
On 03/10/2016 02:05 PM, Matthias Maier wrote:
> @bangerth <https://github.com/bangerth> We could also mention the Tensor
> interface/implementation cleanup in section 2.2 :-)

I think if anything it should have a section of its own. But yes, get on it!

**Comment by bangerth** [comment]
I'm ok with tagging, but I'd still like to understand whether there are things we need to change for Mac/OS X or whether these are just issues on how things are *packaged*, as opposed to things that need to be changed *in the repository*.

**Comment by tamiko** [comment]
> I'm ok with tagging, but I'd still like to understand whether there
> are things we need to change for Mac/OS X or whether these are just
> issues on how things are *packaged*, as opposed to things that need to
> be changed *in the repository*.

As Luca explained some time ago, it is mostly a question of packaging.

Even if it weren't - as opposed to a repository in source form we
provide a fixed, binary mac bundle (I highly doubt that many users use
our packaging scripts to create mac bundles on their own).

**Comment by bangerth** [comment]
OK, then let's go!

What are the missing steps for the release if anyone remembers?

**Comment by luca-heltai** [comment]
Sorry I'm late for the party... Yes, the question is mostly about packaging, however that change (to the repository) would make sure that people compiling their own library don't need to add DYLD_LIBRARY_PATH to their environment anymore, and this is a much cleaner option, in my opinion...

We can go ahead with the tagging, and I'll need some help from @tamiko with cmake to make this work in a clean way.

**Comment by tamiko** [comment]
@luca-heltai Let's get this last fix into the release.

The corresponding configuration is in ```source/CMakeLists.txt```.

**Comment by tamiko** [comment]
@luca-heltai Do we need this upon installation?

**Comment by tjhei** [comment]
okay, release is tagged. Documentation is compiling right now.

**Comment by tjhei** [comment]
@luca-heltai can you please create the .dmg files for 8.4.0?

**Comment by tjhei** [comment]
Thanks, Luca. I am waiting on @guidokanschat uploading the 8.4 docs and will proceed with the announcement directly after.

@bangerth There is nothing else left to do and I have the homepage edits and announcements ready to be pushed online.

**Comment by guidokanschat** [comment]
Documentation is online

https://www.dealii.org/8.4.0/doxygen/deal.II/index.html

Please have a look before we add the link.

**Comment by luca-heltai** [comment]
Yesterday I recompiled the whole set of libraries included in the Mac package with openmpi 1.10.2, due to a bug in the openmpi 1.8.5 version, that was downloaded by default by PETSc. The new package seem to work (thanks to @cox), but I'd be happier if someone else could test this before we proceed with the announcement...
